packages/core/content-manager/admin/src/pages/EditView/components/InputRenderer.tsx

```diff
@@ -35,8 +45,54 @@ type InputRendererProps = DistributiveOmit<EditFieldLayout, 'size'>;
  * the complete EditFieldLayout and will handle RBAC conditions and rendering CM specific
  * components such as Blocks / Relations.
  */
-const InputRenderer = ({ visible, hint: providedHint, ...props }: InputRendererProps) => {
-  const { id, document, collectionType } = useDoc();
+const InputRendererWithContext = (props: InputRendererProps) => {
+  const id = useRelationModalContext('RelationModal', (state) => state.currentRelation.documentId);
+  const document = useRelationModalContext('RelationModal', (state) => state.document);
+  const model = useRelationModalContext('RelationModal', (state) => state.currentRelation.model);
+  const collectionType = useRelationModalContext(
+    'RelationModal',
+    (state) => state.currentRelation.collectionType
+  );
+  const changeCurrentRelation = useRelationModalContext(
+    'RelationModal',
+    (state) => state.changeCurrentRelation
+  );
+  const isModalOpen = useRelationModalContext('RelationModal', (state) => state.isModalOpen);
+  const {
+    edit: { components },
+  } = useDocumentLayout(model);
+
+  return (
+    <InputRenderer
+      contextId={id}
+      contextDocument={document}
+      contextCollectionType={collectionType}
+      contextModel={model}
+      changeCurrentRelation={changeCurrentRelation}
+      isModalOpen={isModalOpen}
+      contextComponents={components}
+      {...props}
+    />
+  );
+};
+
+const InputRenderer = ({
+  visible,
+  hint: providedHint,
+  contextId,
+  contextDocument,
+  contextCollectionType,
+  contextModel,
+  changeCurrentRelation,
+  isModalOpen,
+  contextComponents,
+  ...props
+}: InputRendererProps) => {
+  const { id: hookId, document: hookDocument, collectionType: hookCollectionType } = useDoc();
```

Comment:
I don't think hook is the right diferenciator here. Maybe url? urlId, urlDocument, etc.

Reply:
I agree. I think when the values are coming from `useDoc` we know it is the `root` document. And likewise if it's passed as a prop then it is a _related_ document. Maybe the variable names could delineate this difference?